packages/gatsby-dev-cli/src/watch.js

```diff
@@ -27,53 +27,57 @@ const copyPath = (oldPath, newPath, quiet) =>
   })
 
 function watch(root, packages, { scanOnce, quiet }) {
-  packages.forEach(p => {
-    const prefix = syspath.join(root, `/packages/`, p)
+  const ignored = [
+    /[/\\]node_modules[/\\]/i,
+    /\.git/i,
+    /\.DS_Store/,
+  ].concat(
+    packages.map(p => new RegExp(`${p}[\\/\\\\]src[\\/\\\\]`, `i`))
+  )
+  const watchers = packages.map(p => syspath.join(root, `/packages/`, p))
```

Comment:
Set up an array of packages to watch, rather than iterating with a .forEach (which is kinda janky, and `.on('ready')` will fire for _every_ package.